File: frontend/src/types/material.ts
import type { Session } from "./session";

export type MaterialType = "pdf" | "link" | "video" | "image" | "ppt" | "doc" | "youtube";

export interface Material {
  id: number;
  session_id?: number;
  topic_id?: number;
  title: string;
  description?: string;
  type: MaterialType;
  url: string;
  created_at: string;
  updated_at: string;
}

export interface MaterialInput {
  session_id?: number;
  topic_id?: number;
  title: string;
  description?: string;
  type: MaterialType;
  url: string;
}

export interface SessionWithMaterials extends Session {
  Materials: Material[];
}

File: backend/internal/domain/material.go
package domain

import "time"

type Material struct {
	ID          int       `json:"id"`
	SessionID   *int      `json:"session_id,omitempty"`
	TopicID     *int      `json:"topic_id,omitempty"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Type        string    `json:"type"`
	URL         string    `json:"url"`
	CreatedBy   *string   `json:"created_by,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type MaterialRepository interface {
	Create(material *Material) error
	GetBySessionID(sessionID int) ([]Material, error)
	GetByTopicID(topicID int) ([]Material, error)
	GetByCourseID(courseID int) ([]SessionWithMaterials, error)
	GetByID(id int) (*Material, error)
	Update(material *Material) error
	Delete(id int) error
}
